metadatastore/commands.py

```diff
@@ -65,7 +65,7 @@ def reconfigure(self, config):
     def _connection(self):
         if self.__conn is None:
             self.__conn = MongoClient(self.config['host'],
-                                      self.config.get('port', None))
+                                      int(self.config.get('port', None)))
```

Comment:
I agree with Tom, don't let this leak out of conf because I use the same conf in a few other places and I certainly forget to cast it at times

I like: 
```
int(self.config.get('port', 27017)))
```
less code lol